app/lib/three_scale/redis_config.rb

```diff
@@ -5,10 +5,14 @@ class RedisConfig
     def initialize(redis_config = {})
       raw_config = (redis_config || {}).symbolize_keys
       sentinels = raw_config.delete(:sentinels).presence
+      disable_client_id = raw_config.delete(:disable_client_id).presence
       raw_config.delete_if { |key, value| value.blank? }
 
       @config = ActiveSupport::OrderedOptions.new.merge(raw_config)
       config.sentinels = parse_sentinels(sentinels) if sentinels
+
+      # Disable CLIENT SETNAME, for Redis instances that don't support CLIENT command
+      config.id = nil if disable_client_id
```

Comment:
it might be useful to have a setting `client_id` or rather `client_name` that if set to sometihing will set `config.id` to that value, if false would set `config.id` to `nil`. And if missing or set to `null` will leave default behavior.

Maybe somebody would need setting this to something non-default. And we are adding the option anyway. So maybe it makes sense to add functionality.

Reply:
I don't understand, how does setting id to nil disable SETNAME method?
According to the docs, set-name is disabled by default?? 😕 https://github.com/mperham/sidekiq/wiki/Using-Redis#client-naming

Reply:
Damn, sorry, my bad!

I was confused by [this line](https://github.com/mperham/sidekiq/commit/33846c849a947b1bf9800c22738dadbb953cc49a#diff-96029d2f394258e62375fcff42f91cee34b5e5571bdb099c1f9d82104047c4acR15) - well, because I looked at only this part of code, and not checking the rest. I misinterpreted it, thinking that either the Sidekiq name is set, or it stays nil, but of course I was wrong.

Makes total sense to allow setting a custom ID in our Redis config.

But I am not sure about the default behavior. Is it what you mean @akostadinov ?

`client_name` config | `config.id` | CLIENT SETNAME value
---- | ----- | ----
`custom-name` | `custom-name` | `custom-name` 
`false` | `nil` | -not called-
-missing or null- | -default value set by Sidekiq- | `Sidekiq-#{Sidekiq.server? ? "server" : "client"}-PID-#{$$}`

But allowing false and a string value for `client_name` doesn't seem correct to me.

Reply:
@josemigallas Hmm, good point.
I guess maybe that was changed at some point, I think this is not the behavior in the Sidekiq version porta is using right now. Let me check this!

Reply:
Instead of `false`, it could be empty string `""` also.

Reply:
Re: what @josemigallas pointed out, this default Client naming was indeed removed, since Sidekiq version 6.0.7: https://github.com/mperham/sidekiq/blob/main/Changes.md#607

> Remove Redis connection naming [#4479]

We are using `gem 'sidekiq', '< 6'`

Not sure if that's because of the old Ruby.

@akostadinov @josemigallas and what if we just add `config.id = nil` and be done with it, i.e. no changes in the config, no option to set the `id` explicitly. Just remove it (as this is the default behavior in the future version. And we don't have any feature request to enable this client identification.

So, we just fix the issue for GCP, and not add any new feature.

Reply:
Sounds good. I dunno about version limited, maybe just to prevent unintended upgrade. But we can't quickly update it I guess. At least I don't see it a s a priority.

Reply:
A release note should be added.